string locationA
string locationB
---
uint32 time
